### ISSUE TITLE
Operational Transform Server-Side Example?

### ISSUE THREAD

**Issue by benjismith** [ckeditor/ckeditor5-engine#757]
When ckeditor5 is released, I'll be integrating it in a realtime collaborative writing environment, so I'm very interested in the mechanics of the operational transform implementation.

Do you have any design docs, or a simple prototype, showing how to implement the server-side of the OT system?

**Comment by Reinmar** [comment]
Hi Benji!

Unfortunately, we don't have any (up-to-date) docs yet. We decided to postpone work on them because even the high level ones got outdated pretty quickly and we were wasting time maintaining them. We're going to work on a proper architecture documentation shortly before releasing first alpha version.

Regarding collaboration itself, we've done a lot of research initially and created very rough prototypes (but nothing of a value now – I think it was https://github.com/ckeditor/ckeditor5-design/tree/op-trans, but it's a long time since we did it so I don't even remember if it uses a tree or linear model). One of the most valuable articles was http://www.codecommit.com/blog/java/understanding-and-applying-operational-transformation, but the truth is that this is just the tip of the iceberg. AFAIR, our model is different than this one and requires a bit different approach. Our data model and approach may be pretty unique, so I don't think there will be articles on which you could base.

Besides, as I mentioned, OT itself is just the first step. Its actual application to topics like selections of other users or document metadata requires additional effort.

We're now focused on bringing the first version of CKEditor 5 to life and only after we release it we’ll start publishing more details regarding collaboration. So, please stay tuned :)

PS. I've just found this over the net: https://operational-transformation.github.io/visualization.html. Not sure how well it matches our mechanism, though.

**Comment by interair** [comment]
Hi Reinmar!
Do you have any progress with:
> design docs, or a simple prototype, showing how to implement the server-side of the OT system?

**Comment by Reinmar** [comment]
For those of you who haven't seen the news – we released CKEditor 5 Collaboration Features which you can enable in CKEditor 5 in your application. You can check out the demo in https://docs.ckeditor.com/ckeditor5/latest/features/collaboration/overview.html.

And regarding the docs – I explained why it's unlikely that a complete documentation will ever be available in ckeditor/ckeditor5#759 (comment).
